app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :api_authentication
```

Comment:
This controller is also being used for other existing controllers, to make it more orgaized and more specific for API uses, I think it's better to make a separate `API::Controller`, that way you dont have to skip this method in every controllers that inherit from this controller :)